src/zkbob/ZkBobPool.sol

```diff
@@ -238,16 +243,26 @@ abstract contract ZkBobPool is IZkBobPool, EIP1967Admin, Ownable, Parameters, Zk
         if (txType == 0) {
             // Deposit
             require(transfer_token_delta > 0 && energy_amount == 0, "ZkBobPool: incorrect deposit amounts");
-            IERC20(token).safeTransferFrom(user, address(this), uint256(token_amount) * TOKEN_DENOMINATOR);
+            // Assuming token_amount is in the legacy nomination,
```

Comment:
I don't think we should call this a legacy nomination here and below. I think it's better if we don't have any comment here at all, but instead describe the purpose of numerator constant where it's defined.

Reply:
I will remove comments -- now they were used to highlight these pieces of code that it is important to not forget to change. Originally, I thought that the first iteration of review will contain only the comments to verify the proposed logic with others.